bindings/java/src/org/sleuthkit/datamodel/BlackboardAttribute.java

```diff
@@ -518,8 +521,32 @@ public String getDisplayString() {
 				return Double.toString(getValueDouble());
 			case BYTE:
 				return bytesToHexString(getValueBytes());
-			case DATETIME:
-				return FsContent.epochToTime(getValueLong());
+				
+			case DATETIME: {
+				
+				// return the date/time string in the timezone associated with the datasource,
+				TimeZone tzone = TimeZone.getDefault();
+				
+				try {
+					final Content dataSource = getParentArtifact().getDataSource();
+					if ((dataSource != null) && (dataSource instanceof Image )) {
+						Image  image = (Image) dataSource;
+						tzone = TimeZone.getTimeZone(image.getTimeZone());
+					} 
+				} catch (TskException ex) {
+					// contnue with TimeZone.getDefault();
+				}
+				
+				String timeString = "0000-00-00 00:00:00";
+				long epochSeconds = getValueLong();
+				if (epochSeconds != 0) {
+					synchronized (DATE_FORMATTER) {
+						DATE_FORMATTER.setTimeZone(tzone);
```

Comment:
Might there be another home for epochToTime() functions?

 AbstractFile isn't the the most intuitive p[lace I'd go looking for time conversion utility methods.